homeassistant/components/freebox/sensor.py

```diff
@@ -1,81 +1,135 @@
 """Support for Freebox devices (Freebox v6 and Freebox mini 4K)."""
 import logging
+from typing import Dict
 
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import DATA_RATE_KILOBYTES_PER_SECOND
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
 from homeassistant.helpers.entity import Entity
-
-from . import DATA_FREEBOX
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import (
+    CONNECTION_SENSORS,
+    DOMAIN,
+    SENSOR_DEVICE_CLASS,
+    SENSOR_ICON,
+    SENSOR_NAME,
+    SENSOR_UNIT,
+    TEMPERATURE_SENSOR_TEMPLATE,
+)
+from .router import FreeboxRouter
 
 _LOGGER = logging.getLogger(__name__)
 
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
     """Set up the sensors."""
-    fbx = hass.data[DATA_FREEBOX]
-    async_add_entities([FbxRXSensor(fbx), FbxTXSensor(fbx)], True)
-
-
-class FbxSensor(Entity):
-    """Representation of a freebox sensor."""
-
-    _name = "generic"
-    _unit = None
-    _icon = None
-
-    def __init__(self, fbx):
-        """Initialize the sensor."""
-        self._fbx = fbx
+    router = hass.data[DOMAIN][entry.unique_id]
+    entities = []
+
+    # System sensors
+    syst_datas: Dict[str, any] = await router.system.get_config()
```

Comment:
Why do we need to do this? Can't we just access the stored temperature and connection sensors on the router instance?

Reply:
The router is storing sensors state only, not the name or something else (was the case but changed because of previous review).

BUT, (as written in the comments :blush:) `Name and id of sensors may vary under Freebox devices.`

Reply:
What does `vary` mean here? What consequence does that have?

Reply:
For Freebox Revolution sensors are : 
```json
    "sensors": [
        {
            "id": "temp_hdd",
            "name": "Disque dur",
            "value": 34
        },
        {
            "id": "temp_sw",
            "name": "Température Switch",
            "value": 44
        },
        {
            "id": "temp_cpum",
            "name": "Température CPU M",
            "value": 53
        },
        {
            "id": "temp_cpub",
            "name": "Température CPU B",
            "value": 51
        }
    ],
```

The Freebox Delta or Freebox mini 4K different values since the Delta have more CPU and HDD, the mini 4K does not have a HDD.

So the number of sensors changes and their ids and names.

Reply:
Sorry, I don't understand what the problem is. It doesn't seem to matter that different routers provide different sensors for my initial comment.

Reply:
So I don't understand your initial comment

Reply:
In the sensor.py I init all sensors, on the router I keep sensors status up to date and dispatch the update.

So I need to init sensors the same way as I do in the router.py

Reply:
If we save the sensor dict and not just the sensor values, on the router instance, we don't need to make an additional call here.

Reply:
Yep, but is storing the sensor dict in the router useful ?

Since besides the value, nothing is gonna change.

Will it take more space in the RAM ?

Reply:
Yes, it could take more memory. But it's hard to know how the situation looks without doing memory profiling.

Usually it's wise to avoid doing the same call more than needed. Save the data we need from the first call. When we don't need the data anymore, we can delete it.

Reply:
Will do

Reply:
Done : see b3d67bf9432d7a3ed94a8d9671076435a38115b7